### DISCUSSION TITLE
Classic client removal

### DISCUSSION THREAD

**Discussion by CoBC** [BearWare/TeamTalk5#1919] [Polls]
Hi to everyone,
as you know, since TT5.8, we've stopped developing classic client. Now, thanks to Qt6 and also thanks to some adjustments we made in TT directly, the Qt default client is more accessible. Some little issues are still present but the client is usable without any big issue with screenreaders.
Here, especially in the french community, I see a lot of users with Qt client, there is a minor number of users still using classic.
Also, to be able to use new features, everyone have now to use Qt client.
So the question is: are you OK if we remove classic client now?
Thanks.

**Comment by lbk2907** [comment]
I'm ok with it!

**Comment by lucaschambers** [comment]
please continue to improve and support the classic version since it remains a preferred option due to its better accessibility for screen readers. Despite the changes introduced in the QT version, many users still find it confusing, leading to a continued reliance on the classic version. While there
have been updates to the classic version, it still lacks the advancements seen in version 5.7.0. As such, I urge you to prioritize the upgrading of the classic version to cater to the needs of those who still favor it. Thanks in advance.

**Comment by lucaschambers** [comment]
The classic version is very much smoother and faster compared to QT, and it also provides information on whether an item is expanded or collapsed. Honestly, I am a fan of the classic version and hope that it will not be eliminated.

**Comment by zstanecic** [comment]
the classic version also reports collapsed and expanded state. see lattest version

**Reply by ChrisDuffley** [reply]
Yes, it does, but the Qt also does report collapsed and expanded states in Windows. This does still need to be worked on in Qt on the Mac.

**Comment by CoBC** [comment]
I guess it's a mistake in the reply of @zstanecic, probably he wanted to say that the Qt client also reports item state. Can you confirm?

**Reply by ghost** [reply]
@CoBC 
No. I don't confirm. It only announces when you navigate to a channel name, yes it announces the collapsed or expanded state in that case. 
But you follow this step: Go to a channel which contains sub channels inside, spam the right and left key when the curser is on the channel name. You won't hear "expanded, collapsed, expanded, collapsed" by NVDA. Do the above mentioned with classic version then you will understand it reports collapsed and expanded state continuously and instantly

**Reply by CoBC** [reply]
I know that.
As I said to you by e-mail when you contacted me, it's reported to Qt devs.

**Reply by ghost** [reply]
@CoBC 
the exact list information does not exactly announced by NVDA.
Lets go and describe an example.
team talk 5.7.0 classic: On my server, there are 3 channels, and I'm joined in the root channel. Lets navigate with NVDA: Abolfazl one of 4 level 1, Low internet )(0) 2 of 4 level 1, private channels (0) 3 of 4 level 1, Stereo (0) 4 of 4 level 1.
An example of Team talk 5.13 (windows) default QT: Abolfazl level 1, low internet (0) level 1, private channels (0) level 1, stereo (0) level 1. Why NVDA does not count list information like other windows? An example on my desktop: This PC  1 of 44
This counting should be done by NVDA, not by TeamTalk itself

**Comment by dgleks** [comment]
The TeamTalk QT client, despite the developer's constant assertions of stability, still ships with a significant number of accessibility, UI and performance issues that have yet to be addressed. Some of the more notable ones include:

* Reviewing/navigating messages in the private and channel message windows.
* Tabbing between information boxes in the CTRL+I dialog.
* Sudden, unexplainable UI freezes that lock up screen readers.
* Various longstanding issues with flood protection on servers setup to utilize a shared account (think guest or anonymous).
* Navigating the preference windows/tabs.
* Inability to change sorting order for online users/ban lists, etc.
* Various issues and crashes when attempting to navigate channel ban lists.
* Probably others that I can't think of at the moment.

So, in short, I vote against removing the classic client from the installation package at this time. IMO the classic client shouldn't have even stopped receiving updates until the QT client was made fully accessible and stable, which isn't currently the case.

**Reply by CoBC** [reply]
- Reviewing/navigating messages in the private and channel message windows.: unable to understand without more explanation of the issue, I noticed no issue here
- Tabbing between information boxes in the CTRL+I dialog.: same as previous point
- Sudden, unexplainable UI freezes that lock up screen readers.: you probably referenced freezes when you open some dialogs, so it's already fixed in TT 5.13
- Various longstanding issues with flood protection on servers setup to utilize a shared account (think guest or anonymous).: not understand
- Navigating the preference windows/tabs.: you probably referenced the fact that screen reader announce some combobox items instead of tab title, it's reported to Qt devs and hopefuly will be fix soon
- Inability to change sorting order for online users/ban lists, etc.: you have probably not tested the Qt client for a while, it's possible just using context menu since version 5.9
- Various issues and crashes when attempting to navigate channel ban lists.: can't reproduce

**Reply by ulisesmonge** [reply]
I must say, there's a problem on the bans when you press space to select a banned user, the app completely crashes sometimes. I could record a demo latter.

**Comment by amirmahdifard** [comment]
@CoBC hello, i'm glad you finally got to my point here: i seriously reported all bugs that qt client has, context menu title doesn't announced, colapsd and expanded is not reported automaticly, mooving and selecting more than 1 person to move  to other channel is buggy, screen reader says collapsed when you navigate throo files list items, and some more, and i told you that these issues should be resolved first before removal of classic client, i tryed to move so many of my friends from classic client to qt client and they said qt client has a lot nice abilitys, but we will not switch because it is not properly  accessible, so i hope you fix all qt problems before  you think of removing classic client, also, you still didn't add the move users dialog to qt client.

**Reply by ulisesmonge** [reply]
Hi, moving multiple people on QT client it's easy, it's like Windows explorer. For example, you in the first person from top to botom you start pressing control + down arrow, keeping the control key pressed, and press spacebar on all the people you wish to move, and at the end press control alt x. NVDA will tell you who you're about to move, separated by commas, so you'll know who you will move. I know that QT isn't perfect at accessibility yet, but it's perfectly usable right now and I can do literally everithing on it, that has classic client. So yes, there are some bugs there but it's completely manageable.

**Comment by amirmahdifard** [comment]
@CoBC also,, i agree that classic client is a bit faster than qt client, as others said here and  i voted no because there are still a lot work that qt client has to go throo with accessiblity: then please go notify qt developers so they will fix all bugs more quickly.

**Comment by beqabeqa473** [comment]
Because 64bit applications doesn't give you correct position information

On 6/14/23, shaghayegh1382 ***@***.***> wrote:
> @CoBC
> the exact list information does not exactly announced by NVDA.
> Lets go and describe an example.
> team talk 5.7.0 classic: On my server, there are 3 channels, and I'm joined
> in the root channel. Lets navigate with NVDA: Abolfazl one of 4 level 1, Low
> internet )(0) 2 of 4 level 1, private channels (0) 3 of 4 level 1, Stereo
> (0) 4 of 4 level 1.
> An example of Team talk 5.13 (windows) default QT: Abolfazl level 1, low
> internet (0) level 1, private channels (0) level 1, stereo (0) level 1. Why
> NVDA does not count list information like other windows? An example on my
> desktop: This PC  1 of 44
> This counting should be done by NVDA, not by TeamTalk itself
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/BearWare/TeamTalk5/discussions/1919#discussioncomment-6170964
> You are receiving this because you are subscribed to this thread.
>
> Message ID:
> ***@***.***>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili

**Comment by amirmahdifard** [comment]
@beqabeqa473 i completely didn't understand you in this message

**Comment by lbk2907** [comment]
O, really? I didn’t know that. Thanks for the info.

**Comment by amirmahdifard** [comment]
@lbk2907 that's not a correct info:  an app beeing 32 bit or 64 bit doesn't make difrence with accessibility, that is a qt problem.

**Reply by Nardol** [reply]
NVDA is a 32 bits application, which means there can be issues when getting accessibility information from another 64 bits application.
Consequence is there can be difference in accessibility between 32 and 64 bits applications.

**Reply by CoBC** [reply]
@amirmahdifard Are you confortable with dev or any technical info regarding this topic? It's known by a lot of people that majority of 64 bits applications are concerned by this issue, whether they are Qt or not. So, please don't say that an info is incorrect unless you are absolutely certain of it.

**Comment by lbk2907** [comment]
@amirmahdifard It's working here.

>Hi, moving multiple people on QT client it's easy, it's like Windows explorer. For example, you in the first person from top to botom you start pressing control + down arrow, keeping the control key pressed, and press spacebar on all the people you wish to move, and at the end press control alt x. NVDA will tell you who you're about to move, separated by commas, so you'll know who you will move. I know that QT isn't perfect at accessibility yet, but it's perfectly usable right now and I can do literally everithing on it, that has classic client. So yes, there are some bugs there but it's completely manageable.

@ulisesmonge Thanks for the info.

**Comment by lucaschambers** [comment]
Greetings everyone, I have recently installed the most recent version of Team Talk QT.
However, upon attempting to run teamtalk5.exe, I encountered an error message which I shall include below.
This is one of the significant rationales why I preferred the classic client's retention since it never presented any errors. The error message reads as follows:
"TeamTalk5.exe - Entry Point Not Found dialog
The procedure entry point CreateDXGIFactory2 could not be located in the dynamic link library dxgi.dll.".
OK Button.
If anyone has any knowledge regarding how to solve this problem, Please inform me. Thank you.

**Comment by Amirhossein326** [comment]
Hello. I also disagree with the deletion of teamtalk 5 classic. Because it is easier for me to create a user with the classic version

**Reply by CoBC** [reply]
Hi,
can you explain a bit more? Why it's easier for you to create user with classic?

**Comment by amirmahdifard** [comment]
it's actually easya to create user in qt client
